packages/paypal-commerce-integration/src/paypal-commerce-fastlane/paypal-commerce-fastlane-payment-strategy.ts

```diff
@@ -133,22 +133,27 @@ export default class PaypalCommerceFastlanePaymentStrategy implements PaymentStr
         const state = this.paymentIntegrationService.getState();
         const cartId = state.getCartOrThrow().id;
 
-        const { orderId } = await this.paypalCommerceRequestSender.createOrder(methodId, {
-            cartId,
-        });
+        const isVaultedFlow = paymentData && isVaultedInstrument(paymentData);
 
-        const paymentPayload =
-            paymentData && isVaultedInstrument(paymentData)
-                ? this.prepareVaultedInstrumentPaymentPayload(methodId, orderId, paymentData)
-                : await this.preparePaymentPayload(methodId, orderId, paymentData);
+        try {
+            const paymentPayload = isVaultedFlow
+                ? await this.prepareVaultedInstrumentPaymentPayload(methodId, cartId, paymentData)
+                : await this.preparePaymentPayload(methodId, cartId, paymentData);
```

Comment:
It is not necessary to provide cartId as an option to each method, since you can get it like this:
```
const state = this.paymentIntegrationService.getState();
const cartId = state.getCartOrThrow().id;
```